src/main.rs

```diff
@@ -295,13 +295,48 @@ async fn process_feed(
             .permalink(false)
             .build();
 
-        let rss_item = ItemBuilder::default()
+        let mut rss_item_builder = ItemBuilder::default();
+        rss_item_builder
             .title(title_text)
             .link(base_url.parse(link_url).ok().map(|u| u.to_string()))
             .guid(Some(guid))
             .pub_date(date.map(|date| date.format(&Rfc2822).unwrap()))
-            .description(description)
-            .build();
+            .description(description);
+
+        // Media enclosure
+        if let Some(img_selector) = &config.media {
+            let media = item
+                .as_node()
+                .select_first(&img_selector)
+                .map_err(|()| eyre!("invalid selector for media: {}", img_selector))?;
+
+            let media_attrs = media.attributes.borrow();
+            let media_url = media_attrs
+                .get("src")
+                .ok_or_else(|| eyre!("element selected as image has no 'src' attribute"))?;
+
+            let mut enclosure = Enclosure::default();
+            enclosure.set_url(media_url);
+
+            // Guessing the MIME type from the url as we don't have the full media
+            let parsed_url =
+                Url::parse(media_url).map_err(|e| eyre!("Media enclosure url invalid: {e}"))?;
+            let media_filename = parsed_url
+                .path_segments()
+                .ok_or_else(|| eyre!("Media enclosure url invalid"))?
+                .last()
+                .map(|s| s.to_string())
+                .ok_or_else(|| eyre!("Media enclosure url invalid"))?;
```

Comment:
Since there's a few places where mime detection might fail I suggest the following so that if the selector matches something it will always end up in the feed, even if that means falling back on `application/octet-stream`. (Note below requires importing `mime_guess::mime`):

```suggestion
            let media_mime_type = parsed_url
                .path_segments()
                .and_then(|segments| segments.last())
                .map(|media_filename| mime_guess::from_path(media_filename).first_or_octet_stream())
                .unwrap_or_else(|| mime::APPLICATION_OCTET_STREAM);
            enclosure.set_mime_type(media_mime_type.to_string());
```